app/src/main/java/com/hedvig/app/feature/marketing/pickmarket/PickMarketScreen.kt

```diff
@@ -0,0 +1,294 @@
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.ModalBottomSheetLayout
+import androidx.compose.material.ModalBottomSheetValue
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.material.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.insets.LocalWindowInsets
+import com.hedvig.app.R
+import com.hedvig.app.feature.marketing.PickMarket
+import com.hedvig.app.feature.settings.Language
+import com.hedvig.app.feature.settings.Market
+import com.hedvig.app.ui.compose.composables.buttons.LargeContainedButton
+import com.hedvig.app.ui.compose.theme.HedvigTheme
+import kotlinx.coroutines.launch
+
+enum class PickMarketSheet {
+    MARKET,
+    COUNTRY,
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun PickMarketScreen(
+    onSubmit: () -> Unit,
+    onSelectMarket: (Market) -> Unit,
+    onSelectLanguage: (Language) -> Unit,
+    data: PickMarket
+) {
+    var sheet by rememberSaveable { mutableStateOf<PickMarketSheet?>(null) }
+    val coroutineScope = rememberCoroutineScope()
+    val modalBottomSheetState = rememberModalBottomSheetState(ModalBottomSheetValue.Hidden)
+
+    val insets = LocalWindowInsets.current
+    val navigationBarHeight = with(LocalDensity.current) { insets.navigationBars.bottom.toDp() }
+    ModalBottomSheetLayout(
+        sheetState = modalBottomSheetState,
+        sheetContent = {
```

Comment:
Maybe extract the sheet content to another composable? Kind of long with this and the screen content beeing in the same composable

Reply:
I did consider this, however, I felt like passing all of the dependencies to this composable would be almost as long as the composable itself, making it slightly pointless to extract it 😞